### DISCUSSION TITLE
Negative values of undefined metabolite in blood cell compartment of small intestine

### DISCUSSION THREAD

**Discussion by Bcicali** [Open-Systems-Pharmacology/Forum#1398] [Questions & Problems] [answered]
Hi All,

I am running into this strange simulation run failure where an undefined CYP1A2 metabolite is becoming negative in the blood cells within the lower jejunum. The amount of metabolite should theoretically be nearly 0, and allowing this molecular parameter to be negative solves the simulation run failure issue. However, the workflow of sending each and every simulation to MoBi to allow the parameter to be negative and sending back to PK-Sim before running development and validation simulations is quite cumbersome. So I was wondering if 1) is there a way to allow negative values for this particular parameter within PK-Sim, or 2) figure out why sometimes such a small amount of metabolite in blood cells of the lower jejunum is throwing this error in the first place?

Thanks!
Brian

**Comment by StephanSchaller** [comment, marked as answer]
Hi Brian, this is likely due to numeric noise. Did you try adjusting the solver settings (tolerances)?

**Reply by Bcicali** [reply]
Hi Stephan,

I have the absolute tolerance set to 1E-30, with a relative tolerance at 1E-05. However, still had the issue with some of the simulations. I was able to circumvent the issue using population simulations to avoid the numeric noise failing the simulation, but is there a better solver setting fix? i.e. further increase solver tolerance?

**Comment by PavelBal** [comment]
@Bcicali Hi Brian,

general notes on tolerances. Try to keep the absolute and the relative tolerance values close together, e.g. 1e-9 and 1e-7. Best to reduce them together in one step. 1e-30 is an insane value, I am surprised it works at all :D From my experience, everything below 1e-15 for the absolute tolerance makes things even worse.

**Reply by Bcicali** [reply]
Hi Pavel,

Thank you very much for the insight, I will keep this in mind moving forward ;)
